deps/triemap/triemap.h

```diff
@@ -164,8 +166,12 @@ void TrieMapIterator_Free(TrieMapIterator *it);
 int TrieMapIterator_Next(TrieMapIterator *it, char **ptr, tm_len_t *len, void **value);
 
 /* Iterate to the next matching entry in the trie. Returns 1 if we can continue,
- * or 0 if we're done and should exit */
+ * or 0 if we're done and should exit 
+ * NextContains is used by Contains and Suffix queries.
+ * Wildcard is used by Wildcard queries.
+ */
 int TrieMapIterator_NextContains(TrieMapIterator *it, char **ptr, tm_len_t *len, void **value);
+int TrieMapIterator_NextWildcard(TrieMapIterator *it, char **ptr, tm_len_t *len, void **value);
```

Comment:
How will you know that a timeout happened? You will not get it from the return value ...

Reply:
That is true for all timeouts we may suffer during the build of the iteration tree.
We do check for timeout once `QAST_Iterate` call is done and if true, set status to `QUERY_TIMEDOUT`.